File: test/Data.Analyzers.Vsix.Test/CSharp/ProjectionColumnNaming.cs
namespace DevZest.Data.Analyzers.Vsix.Test.CSharp
{
    public class ProjectionColumnNaming : Model
    {
        protected static readonly Mounter<_Int32> _Column1 = RegisterColumn((ProjectionColumnNaming _) => _.Column1);

        public _Int32 Column1 { get; private set; }

        public class Lookup : Projection
        {
            static Lookup()
            {
                Register((Lookup _) => _.Column2, _Column1);
            }

            public _Int32 Column2 { get; private set; }
        }
    }
}

File: test/Data.Analyzers.Vsix.Test/VisualBasic/ProjectionColumnNaming.vb
Public Class ProjectionColumnNaming
    Inherits Model

    Protected Shared ReadOnly _Column1 As Mounter(Of _Int32) = RegisterColumn(Function(x As ProjectionColumnNaming) x.Column1)

    Private m_Column1 As _Int32
    Public Property Column1 As _Int32
        Get
            Return m_Column1
        End Get
        Private Set
            m_Column1 = Value
        End Set
    End Property

    Public Class Lookup
        Inherits Projection

        Shared Sub New()
            Register(Function(x As Lookup) x.Column2, _Column1)
        End Sub

        Private m_Column2 As _Int32
        Public Property Column2 As _Int32
            Get
                Return m_Column2
            End Get
            Private Set
                m_Column2 = Value
            End Set
        End Property
    End Class
End Class
